File: java/lasagna/Iteration1/src/main/java/Lasagna.java
public class Lasagna {

    private static final int MINUTES_PER_LAYER = 2;
    
    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int minutesInOven) {
        return expectedMinutesInOven() - minutesInOven; 
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return MINUTES_PER_LAYER * numberOfLayers;
    }

    public int totalTimeInMinutes(int numberOfLayers, int minutesInOven) {
        return preparationTimeInMinutes(numberOfLayers) + minutesInOven;
    }
}

File: python/guidos-gorgeous-lasagna/Iteration3/lasagna.py
"""
Calculates the cook time for a lasagna.
"""

EXPECTED_BAKE_TIME = 40
MINUTES_PER_LAYER = 2

def bake_time_remaining(elapsed_bake_time):
    """Calculate the bake time remaining.

    :param elapsed_bake_time: int - baking time already elapsed.
    :return: int - remaining bake time (in minutes) derived from 'EXPECTED_BAKE_TIME'.

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    """
    return EXPECTED_BAKE_TIME - elapsed_bake_time 


def preparation_time_in_minutes(number_of_layers):
    """Calculate the preparation time in minutes.

    :param number_of_layers: int - the number of layers to prepare.
    :return: int - the preparation time in minutes.

    Function that takes the number of layers and returns the total preparation time in minutes.
    """
    return number_of_layers * MINUTES_PER_LAYER


def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """Calculate the elapsed time in minutes.

    :param number_of_layers: int - the number of layers to prepare.
    :param elapsed_bake_time: int - the time that has elapsed so far.
    :return: int - the elapsed time in minutes.
    """
    return preparation_time_in_minutes(number_of_layers) + elapsed_bake_time